gradio/components/file.py

```diff
@@ -125,6 +126,15 @@ def __init__(
     def _process_single_file(self, f: FileData) -> NamedString | bytes:
         file_name = f.path
         if self.type == "filepath":
+            mime_type = client_utils.get_mimetype(file_name)
+            if (
+                self.file_types
+                and mime_type
+                and not client_utils.is_valid_file(mime_type, self.file_types)
+            ):
+                raise Error(
+                    f"Invalid file type: {mime_type}. Please upload a file that is one of these formats: {self.file_types}"
+                )
```

Comment:
Looks good @dawoodkhan82. nit, but there is some code duplication, thoughts on abstracting this part into a function and calling that function from the four components?

Reply:
Or I guess you could even just move some of this into `is_valid_file`?

Reply:
Concretely, I'd suggest moving the `mime_type = client_utils.get_mimetype(file_name)` into the `is_valid_file` function to reduce the duplication

Reply:
Yeah good point